### PR TITLE
Fixes to the annotation tool, json schema and spec to be in sync with new grammar

### PR DESCRIPTION
**Pull request by kavyasrinet**
# Description

This PR has patches for pieces that were out of sync in the current grammar caused by transition of ownership 
1. `has_block_type` is now in triples
2. `dance_type_name` is now in triples (`has_tag`)
3. `quantity` is now a dictionary : `'quantity': {'attribute': <>}`
4. `ordinal` is now just a span compared to `"ordinal": {"ordinal_span": <>}`
5. `'return_quantity'` can be a span, added to markdown
6. `dialogue_target` is now in json schema
7. A corner case bug where if people click on multiple relative directions and select reference object and then go back and forth was getting recorded in state change in backend and was resulting into weird dictionaries has been resolved here.
8. A bug was introduced in the transition that was meddling with relative direction "BETWEEN" having `reference_object_1` and `reference_object_2` has been fixed.
9. fix `location : SPEAKER_LOOK` to `location: {'reference_object': {'special_reference': 'SPEAKER_LOOK'}}`
10. `reference_object_1` and `reference_object_2` to have filters
11. `action_type: COPY` -> `action_type: BUILD`

Please note that this PR also contains training script changes from @snyxan that shouldn't yet be checked in .

## Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [x] I want a high level review. 
- [ ] I want a deep design review.

# Checklist:

- [ ] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I have verified that all scripts in `tests/scripts` runs on hardware.

### CODE REVIEW

File: droidlet/perception/semantic_parsing/nsp_transformer_model/caip_dataset.py [hunk 1]
```diff
@@ -120,6 +120,12 @@ def __getitem__(self, idx):
             p_text, p_tree = t
         except ValueError as e:
             print(e)
+        try:
+            t = self.data[dtype][idx % len(self.data[dtype])]
+            p_text, p_tree = t
+        except ValueError as e:
+            print(e)
+        print(f"ptext: {p_text}, ptree: {p_tree}")
```
Comment (snyxan) [inline droidlet/perception/semantic_parsing/nsp_transformer_model/caip_dataset.py:128]: should remove this

File: droidlet/perception/semantic_parsing/nsp_transformer_model/utils_model.py [hunk 2]
```diff
@@ -26,7 +26,10 @@ def build_model(args, tree_i2w):
 
 
 def load_model(model_dir, model_name="caip_test_model"):
-    path = os.path.join(model_dir, model_name + ".pth")
+    if model_name[-3:] == "pth":
```
Comment (snyxan) [inline droidlet/perception/semantic_parsing/nsp_transformer_model/utils_model.py:29]: should remove this

File: tools/nsp_scripts/train_model.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import os
```
Comment (snyxan) [inline tools/nsp_scripts/train_model.py:9]: should remove all the changes in this file

**Review by snyxan — APPROVED** [review]
On high level LGTM. We'll test it during dataset validation and maybe push more fixes when we see more errors.
One thing is we should remove the changes related to nsp model training (later we should fix it all and nuke the `tools/nsp_scripts` folder).

File: droidlet/perception/semantic_parsing/nsp_transformer_model/caip_dataset.py (hunk 1 above)
Comment (kavyasrinet) [reply droidlet/perception/semantic_parsing/nsp_transformer_model/caip_dataset.py]: whoops you reviewed this before I removed these files. Could you review again ?

File: droidlet/perception/semantic_parsing/nsp_transformer_model/utils_model.py (hunk 2 above)
Comment (kavyasrinet) [reply droidlet/perception/semantic_parsing/nsp_transformer_model/utils_model.py]: same as above

File: tools/nsp_scripts/train_model.py (hunk 3 above)
Comment (kavyasrinet) [reply tools/nsp_scripts/train_model.py]: same as above

**Comment by snyxan** [comment]
About the two new commits:
aecc5e4 looks good to me.
for 37fd31e, I wasn't expecting these three files ( `droidlet/perception/semantic_parsing/nsp_transformer_model/caip_dataset.py`, `droidlet/perception/semantic_parsing/nsp_transformer_model/utils_model.py` and `tools/nsp_scripts/train_model.py`) to be deleted, just restore them back to the original state?

**Comment by kavyasrinet** [comment]
Uh you are right - the files shouldn't have been deleted, just restored to their respective versions in main. Let me dig into what's going on

**Comment by kavyasrinet** [comment]
This should be fixed now, can you double check @snyxan

**Comment by snyxan** [comment]
LGTM!

**Pull request merged**